pandas/core/apply.py

```diff
@@ -307,23 +306,36 @@ def apply_standard(self):
                 # reached via numexpr; fall back to python implementation
                 pass
             else:
-                return self.obj._constructor_sliced(result, index=labels)
+                # this means that the reduction was successful
+                if partial_result is None:
+                    return self.obj._constructor_sliced(result, index=labels)
+                else:
+                    if isinstance(partial_result, Series):
```

Comment:
I think it would be better to get this done in the `compute_reduction` however i am not sure how to do it in the best way.